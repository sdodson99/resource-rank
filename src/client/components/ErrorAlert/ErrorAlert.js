import React from 'react';
import PropTypes from 'prop-types';
import styles from './ErrorAlert.module.css';
import Alert from '../Alert/Alert';
import { Error as ErrorIcon } from '@material-ui/icons';

const ErrorAlert = ({ children, border, scrollTo }) => (
  <div className={styles.ErrorAlert} data-testid="ErrorAlert">
    <Alert
      border={border}
      scrollTo={scrollTo}
      className="text-red-700 border-red-200 bg-red-50"
      icon={<ErrorIcon />}
    >
      {children}
    </Alert>
  </div>
);

ErrorAlert.propTypes = {
  children: PropTypes.node,
  border: PropTypes.bool,
  scrollTo: PropTypes.bool,
};

ErrorAlert.defaultProps = {};

export default ErrorAlert;
