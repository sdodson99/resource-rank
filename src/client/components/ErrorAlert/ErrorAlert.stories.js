import React from 'react';
import ErrorAlert from './ErrorAlert';

export default {
  component: ErrorAlert,
  title: 'Components/ErrorAlert',
  decorators: [
    (Story) => {
      return <Story />;
    },
  ],
};

const Template = (args) => <ErrorAlert {...args} />;

export const Default = Template.bind({});
Default.args = {
  children: 'Something went wrong.',
};

export const Border = Template.bind({});
Border.args = {
  ...Default.args,
  border: true,
};
