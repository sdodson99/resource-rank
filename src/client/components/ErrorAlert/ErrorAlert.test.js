import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ErrorAlert from './ErrorAlert';
import { createRenderer } from 'react-test-renderer/shallow';

describe('<ErrorAlert />', () => {
  test('it should mount', () => {
    render(<ErrorAlert />);

    const errorAlert = screen.getByTestId('ErrorAlert');

    expect(errorAlert).toBeInTheDocument();
  });

  it('should render correctly', () => {
    const page = createRenderer().render(<ErrorAlert>Message</ErrorAlert>);

    expect(page).toMatchSnapshot();
  });

  describe('with border', () => {
    it('should render correctly', () => {
      const page = createRenderer().render(
        <ErrorAlert border={true}>Message</ErrorAlert>
      );

      expect(page).toMatchSnapshot();
    });
  });
});
