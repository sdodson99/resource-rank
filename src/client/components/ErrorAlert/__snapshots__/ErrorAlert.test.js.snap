// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorAlert /> should render correctly 1`] = `
<div
  className="ErrorAlert"
  data-testid="ErrorAlert"
>
  <Alert
    className="text-red-700 border-red-200 bg-red-50"
    icon={<Memo />}
  >
    Message
  </Alert>
</div>
`;

exports[`<ErrorAlert /> with border should render correctly 1`] = `
<div
  className="ErrorAlert"
  data-testid="ErrorAlert"
>
  <Alert
    border={true}
    className="text-red-700 border-red-200 bg-red-50"
    icon={<Memo />}
  >
    Message
  </Alert>
</div>
`;
